pkg/kubelet/volumemanager/reconciler/reconciler_common.go

```diff
@@ -158,8 +158,16 @@ func (rc *reconciler) Run(stopCh <-chan struct{}) {
 }
 
 func (rc *reconciler) unmountVolumes() {
-	// Ensure volumes that should be unmounted are unmounted.
+	mountedVolumes := []cache.MountedVolume{}
 	for _, mountedVolume := range rc.actualStateOfWorld.GetAllMountedVolumes() {
+		if rc.operationExecutor.IsOperationPending(mountedVolume.VolumeName, mountedVolume.PodName, nestedpendingoperations.EmptyNodeName) {
```

Comment:
why checking IsOperationPending in a sapearte for loop instead of inside the original loop?

Reply:
Will the following work?
for each volumes 
1. check operation pending, if yes, skip
2. (no pending operation) check the volume is still mounted or not again, and then move on with unmount if it is still mounted

Reply:
> why checking IsOperationPending in a sapearte for loop instead of inside the original loop?

Yeah, this is somewhat distracting refactoring

Reply:
This change could not solve the umount race,  assume that at line `rc.actualStateOfWorld.GetAllMountedVolumes()`  the volume is still umounting in nested_operations_goroutine,  but at line`  if rc.operationExecutor.IsOperationPending(mountedVolume.VolumeName, mountedVolume.PodName, nestedpendingoperations.EmptyNodeName)` the nested_operations_goroutine just finished,   the volume would still umount twice,  so we should also check `if rc.actualStateOfWorld.PodRemovedFromVolume(mountedVolume.PodName, mountedVolume.VolumeName)` before call umountVolume for every volume

Reply:
This change could not solve the umount race, assume that at line `rc.actualStateOfWorld.GetAllMountedVolumes() ` the volume is still umounting in nested_operations_goroutine, but at line `if rc.operationExecutor.IsOperationPending(mountedVolume.VolumeName, mountedVolume.PodName, nestedpendingoperations.EmptyNodeName) ` the nested_operations_goroutine just finished, the volume would still umount twice, so we should also check if `rc.actualStateOfWorld.PodRemovedFromVolume(mountedVolume.PodName, mountedVolume.VolumeName) `before call umountVolume for every volume



```
	for _, mountedVolume := range rc.actualStateOfWorld.GetAllMountedVolumes() {
		if !rc.desiredStateOfWorld.PodExistsInVolume(mountedVolume.PodName, mountedVolume.VolumeName) &&
			!rc.operationExecutor.IsOperationPending(mountedVolume.VolumeName, mountedVolume.PodName, nestedpendingoperations.EmptyNodeName) &&
			!rc.actualStateOfWorld.PodRemovedFromVolume(mountedVolume.PodName, mountedVolume.VolumeName) {
			// Volume is mounted, unmount it
			klog.V(5).InfoS(mountedVolume.GenerateMsgDetailed("Starting operationExecutor.UnmountVolume", ""))
			err := rc.operationExecutor.UnmountVolume(
```